src/vaults/LockedWithdrawal.sol

```diff
@@ -0,0 +1,151 @@
+     * @dev debtShares = token_to_withdraw * price of token
+     * @dev user will get the share of debtShare after selling earn token
+     */
+    function registerWithdrawalRequest(address user, uint256 debtShares) external onlyVault {
+        _mint(user, debtShares);
+        requestTimes[user] = block.timestamp;
+        pendingDebtShares += debtShares;
+    }
+
+    /**
+     * @notice Resolve the pending debt token after closing a position
+     * @param resolvedAmount amount resolved after pay token to this contract.
+     * @dev This will increase the amount of paytoken and total supply of debt token in the pool
+     * @dev More user will be allowed to withdraw funds
+     * @dev the resolved amount is the ratio of locked e-earn token and minimun e earn token available to burn in vault.
+     * @dev resolvedAmount = pendinDebtToken * min(vault_available_e_earn_to_burn, locked_e_earn) / locked_e_earn
+     */
+    function resolveDebtShares(uint256 resolvedAmount) external onlyVault {
+        pendingDebtShares -= resolvedAmount;
+    }
+
+    /**
+     * @notice calculate the total resolved debt share
+     * @return total resoved debt share
+     */
+    function getResolvedShares() internal view returns (uint256) {
+        // total share is total token supply  - not resolved debt token
+        return totalSupply - pendingDebtShares;
+    }
+
+    /**
+     * @notice Release all the available funds of the user.
+     * @return tokenShare amount of asset user gets
+     * @dev required to have enough share in debt token, As we dont have cancellation policy.
+     * @dev user will get the full amount proportion of debtShare
+     */
+    function redeem() external returns (uint256) {
+        // check for sla
+        require(block.timestamp > requestTimes[msg.sender] + sla, "LWE: before SLA time");
+
+        uint256 resolvedShares = getResolvedShares();
+
+        // debt token share
+        uint256 userShares = balanceOf[msg.sender];
+
+        // check if the user share is resolved
+        require(userShares <= resolvedShares, "LWE: Unresolved debts");
+
+        // total token supply for payment
+        uint256 totalAssets = asset.balanceOf(address(this));
+
+        // amount of asset to pay to user
+        uint256 assetsToUser = totalAssets.mulDivDown(userShares, resolvedShares);
+
+        // transfer the amount.
+        asset.safeTransfer(msg.sender, assetsToUser);
+        // burn the user token.
+        _burn(msg.sender, userShares);
+
+        return assetsToUser;
+    }
+
+    ///////////////////////////////////////
+    // View for the user / fontend
+    //////////////////////////////////////
+
+    /**
+     * @notice check if user can withdraw funds now.
+     * @return true or false if user can withdraw the funds or not
+     */
+    function canWithdraw(address user) public view returns (bool) {
+        if (block.timestamp < requestTimes[user] + sla) {
```

Comment:
Since you use `block.timestamp > requestTimes[msg.sender] + sla` in `redeem` I think you have to use `<=` here.